src/Microsoft.AspNetCore.Cors/Infrastructure/CorsPolicyBuilder.cs

```diff
@@ -144,6 +156,17 @@ public CorsPolicyBuilder AllowAnyHeader()
         }
 
         /// <summary>
+        /// Enables the policy to be configured with wildcarded domains that will then allow
+        /// CORS from any matching subomdain
+        /// </summary>
+        /// <returns>The current policy builder</returns>
+        public CorsPolicyBuilder AllowWildcardSubdomains()
+        {
+            _policy.IsOriginAllowed = _policy.IsOriginAnAllowedSubdomain;
```

Comment:
Related to the above comment, this method is also *destructive*. Would it make more sense to have some way of *combining* the `Func<>`s? That would certainly be more flexible, though perhaps too complex? Thoughts?